qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,698 @@
+                if key in self._params:
+                    if parameter_not_found:
+                        candidates = self._params[key]
+                        parameter_not_found = False
+                    else:
+                        raise CalibrationError("Duplicate parameters.")
+
+        # 4) Filter candidate parameter values.
+        if valid_only:
+            candidates = [val for val in candidates if val.valid]
+
+        candidates = [val for val in candidates if val.group == group]
+
+        if cutoff_date:
+            candidates = [val for val in candidates if val.date_time <= cutoff_date]
+
+        if len(candidates) == 0:
+            msg = f"No candidate parameter values for {param_name} in calibration group {group} "
+
+            if qubits:
+                msg += f"on qubits {qubits} "
+
+            msg += f"in schedule {sched_name}"
+
+            if cutoff_date:
+                msg += f" Cutoff date: {cutoff_date}"
+
+            raise CalibrationError(msg)
+
+        # 5) Return the most recent parameter.
+        candidates.sort(key=lambda x: x.date_time)
+
+        return candidates[-1].value
+
+    def get_schedule(
+        self,
+        name: str,
+        qubits: Tuple[int, ...],
+        free_params: List[str] = None,
+        group: Optional[str] = "default",
+    ) -> Schedule:
+        """
+        Get the schedule with the non-free parameters assigned to their values.
+
+        Args:
+            name: The name of the schedule to get.
+            qubits: The qubits for which to get the schedule.
+            free_params: The parameters that should remain unassigned.
+            group: The calibration group from which to draw the
+                parameters. If not specifies this defaults to the 'default' group.
+
+        Returns:
+            schedule: A copy of the template schedule with all parameters assigned
+                except for those specified by free_params.
+
+        Raises:
+            CalibrationError: if the name of the schedule is not known or if a parameter could
+                not be found.
+        """
+        if (name, qubits) in self._schedules:
+            schedule = self._schedules[(name, qubits)]
+        elif (name, None) in self._schedules:
+            schedule = self._schedules[(name, None)]
+        else:
+            raise CalibrationError(f"Schedule {name} is not defined for qubits {qubits}.")
+
+        # Retrieve the channel indices based on the qubits and bind them.
+        binding_dict = {}
+        for ch in schedule.channels:
+            if ch.is_parameterized():
+                binding_dict[ch.index] = self._get_channel_index(qubits, ch)
+
+        # Binding the channel indices makes it easier to deal with parameters later on
+        schedule = schedule.assign_parameters(binding_dict, inplace=False)
+
+        # The following code allows us to get the keys when the schedule has call instructions.
```

Comment:
You don't need this. Seem my comment below.

Reply:
That has been taken care of.